dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Transformers/BaseClasses.mo

```diff
@@ -98,142 +91,77 @@ partial model BaseTapChangerPhaseShifter_MAX "Base model for tap-changers and ph
 
 equation
 
-  when (valueToMonitor.value > valueMax)  then
+  when (valueToMonitor.value > valueMax) and not(locked) then
     valueAboveMax = true;
+    tValueAboveMaxWhileRunning = time;
     valueUnderStop = false;
     Timeline.logEvent1(TimelineKeys.PhaseShifterAboveMax);
-  elsewhen (valueToMonitor.value <= valueStop) then
+  elsewhen (valueToMonitor.value <= valueStop) and not(locked) then
     valueAboveMax = false;
+    tValueAboveMaxWhileRunning = Constants.inf;
     valueUnderStop = true;
     Timeline.logEvent1(TimelineKeys.PhaseShifterBelowStop);
+   elsewhen running.value and locked then
+    valueAboveMax = pre(valueAboveMax);
+    tValueAboveMaxWhileRunning = Constants.inf;
+    valueUnderStop = pre(valueUnderStop);
   end when;
 
   lookingToDecreaseTap = running.value and valueAboveMax and decreaseTapToDecreaseValue;
   lookingToIncreaseTap = running.value and valueAboveMax and increaseTapToDecreaseValue;
 
-  //The "not(locked)" condition is important to prevent unwanted transitions when the phase-shifter leaves the "locked" state
-  when running.value and valueAboveMax and not(locked) then
-    tValueAboveMaxWhileRunning= time;
-  elsewhen running.value and valueUnderStop and not(locked) then
-    tValueAboveMaxWhileRunning= Constants.inf;
-  elsewhen running.value and locked then
-    tValueAboveMaxWhileRunning= Constants.inf;
-  end when;
-
-  //Transition to "locked"
+  //Transition to "Locked" (possible from any state and prioritary)
   when (not running.value) or locked then
     state = State.Locked;
     tap.value = pre(tap.value);
     tTapUp = Constants.inf;
     tTapDown = Constants.inf;
-  //Transition from "locked" to "standard"
-  elsewhen running.value and not(locked) and not(valueAboveMax) and pre(state) == State.Locked then
-    state = State.Standard;
-    tap.value = pre(tap.value);
-    tTapUp = Constants.inf;
-    tTapDown = Constants.inf;
-  //Transition from "locked" to "waitingToMoveDown"
-  elsewhen running.value and not(locked) and lookingToDecreaseTap and pre(state) == State.Locked then
+  //Transition to "WaitingToMoveDown" (possible from any state except down states)
+  elsewhen lookingToDecreaseTap and (pre(state) == State.Standard or pre(state) == State.MoveUp1 or pre(state) == State.MoveUpN or pre(state) == State.WaitingToMoveUp or pre(state) == State.Locked) then
```

Comment:
the test on running and not(locked) disappeared in the when conditions, where they useless?

Reply:
I think so but I would double check because it is always difficult not to miss something

Reply:
Corrected and check on IEEE14.